sql/core/src/main/scala/org/apache/spark/status/protobuf/sql/SQLExecutionUIDataSerializer.scala

```diff
@@ -23,17 +23,17 @@ import collection.JavaConverters._
 
 import org.apache.spark.sql.execution.ui.SQLExecutionUIData
 import org.apache.spark.status.protobuf.{JobExecutionStatusSerializer, ProtobufSerDe, StoreTypes}
-import org.apache.spark.status.protobuf.Utils.getOptional
+import org.apache.spark.status.protobuf.Utils._
 
 class SQLExecutionUIDataSerializer extends ProtobufSerDe[SQLExecutionUIData] {
 
   override def serialize(ui: SQLExecutionUIData): Array[Byte] = {
     val builder = StoreTypes.SQLExecutionUIData.newBuilder()
     builder.setExecutionId(ui.executionId)
     builder.setRootExecutionId(ui.rootExecutionId)
-    Option(ui.description).foreach(builder.setDescription)
-    Option(ui.details).foreach(builder.setDetails)
-    Option(ui.physicalPlanDescription).foreach(builder.setPhysicalPlanDescription)
+    setStringField(ui.description, builder.setDescription)
```

Comment:
A strange compilation error with Java 11 and 17. Do you have any suggestions? @gengliangwang 

```
[ERROR] [Error] /Users/yangjie01/SourceCode/git/spark-mine-12/sql/core/src/main/scala/org/apache/spark/status/protobuf/sql/SQLExecutionUIDataSerializer.scala:34: type mismatch;
 found   : org.apache.spark.status.protobuf.StoreTypes.SQLExecutionUIData.Builder
 required: com.google.protobuf.MessageOrBuilder

```

Reply:
Hmmm...I have java 11 on local but don't have this issue. We can change the type from `com.google.protobuf.MessageOrBuilder` to `Any` for a workaround if you can't bypass it.

Reply:
GA can't be compiled with Java 11 and 17. Then I  can't be compiled locally too with Java 11:

```
yangjie01@B000000396921U spark-mine-12 % java -version
openjdk version "11.0.17" 2022-10-18 LTS
OpenJDK Runtime Environment Zulu11.60+19-CA (build 11.0.17+8-LTS)
OpenJDK 64-Bit Server VM Zulu11.60+19-CA (build 11.0.17+8-LTS, mixed mode)
```

Reply:
`Any` can build successful

Reply:
[1a7fd09](https://github.com/apache/spark/pull/39682/commits/1a7fd0976b83666b4e47a37cb527d8f5874ba1b9) change to use `Any` instead of `com.google.protobuf.MessageOrBuilder` for test maven build

Reply:
Not related to Java version. Calling `setStringField` in other modules will make maven build fail, change to `Any` can build success,, give an independent pr: https://github.com/apache/spark/pull/39688